### PR TITLE
Removes redundant buffer zeroing in NSStringAPI

### PR DESCRIPTION
**Pull request by valeriyvan**

### CODE REVIEW

**Comment by milseman** [comment]
@ianpartridge @millenomi, do you know who would be a good person to review and integrate this PR?

**Comment by spevans** [comment]
@swift-ci test linux

File: Sources/Foundation/NSStringAPI.swift [hunk 1]
```diff
@@ -74,9 +74,12 @@ internal func _persistCString(_ p: UnsafePointer<CChar>?) -> [CChar]? {
...
     let len = UTF8._nullCodeUnitOffset(in: cString)
-    var result = [CChar](repeating: 0, count: len + 1)
-    for i in 0..<len {
-        result[i] = cString[i]
+    let result = [CChar](unsafeUninitializedCapacity: len + 1) { buf, initedCount in
+        for i in 0..<len {
+            buf[i] = cString[i]
+        }
+        buf[len] = 0
+        initedCount = len + 1
```
Comment (spevans) [inline Sources/Foundation/NSStringAPI.swift]: Would the copy be better as:
```
let bytesToCopy = UTF8._nullCodeUnitOffset(in: cString) + 1 // +1 for the terminating NUL
let result = [CChar](unsafeUninitializedCapacity: bytesToCopy) { buf, initedCount in
        buf.baseAddress!.assign(from: cString, count: bytesToCopy)
         initedCount = bytesToCopy
}
```

Comment (valeriyvan) [reply Sources/Foundation/NSStringAPI.swift]: You are right. Thank you for suggestion.

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by spevans** [comment]
@swift-ci test linux

**Pull request merged**